tests/test_cs50.py

```diff
@@ -0,0 +1,151 @@
+import math
+import sys
+import unittest
+
+from unittest.mock import patch
+
+from cs50.cs50 import get_string, _get_int, _get_float
+
+
+class TestCS50(unittest.TestCase):
+    @patch("cs50.cs50._get_input", return_value="")
+    def test_get_string_empty_input(self, mock_get_input):
+        """Returns empty string when input is empty"""
+        self.assertEqual(get_string("Answer: "), "")
+        mock_get_input.assert_called_with("Answer: ")
+
+
+    @patch("cs50.cs50._get_input", return_value="test")
+    def test_get_string_nonempty_input(self, mock_get_input):
+        """Returns the provided non-empty input"""
+        self.assertEqual(get_string("Answer: "), "test")
+        mock_get_input.assert_called_with("Answer: ")
+
+
+    @patch("cs50.cs50._get_input", side_effect=EOFError)
+    def test_get_string_eof(self, mock_get_input):
+        """Returns None on EOF"""
+        self.assertIs(get_string("Answer: "), None)
+        mock_get_input.assert_called_with("Answer: ")
+
+
+    def test_get_string_invalid_prompt(self):
+        """Raises TypeError when prompt is not str"""
+        with self.assertRaises(TypeError):
+            get_string(1)
+
+
+    @patch("cs50.cs50.get_string", return_value=None)
+    def test_get_int_eof(self, mock_get_string):
+        """Returns None on EOF"""
+        self.assertIs(_get_int("Answer: "), None)
```

Comment:
How does this know to send EOF?

Reply:
Since we're just testing `get_int` here, we can assume that `get_string` is working correctly. In this particular test, we're assuming that `get_string` returns `None` on EOF, which we know should be the case. And so the `patch` decorator here replaces `get_string` with a mock implementation that just returns `None`. And the test is just making sure that `get_int` also returns `None` in this case.